src/haddock/gear/prepare_run.py

```diff
@@ -1079,7 +1104,7 @@ def populate_mol_parameters(modules_params: ParamMap) -> None:
         Alter the dictionary in place.
     """
     # the starting number of the `mol_` parameters is 1 by CNS definition.
-    num_mols = range(1, len(modules_params["topoaa.1"]["molecules"]) + 1)
+    num_mols = range(1, len(topology_params["molecules"]) + 1)
```

Comment:
Why can't you just re-use `modules_params` here?

Reply:
Because now molecules are stored in the `topology_params["molecules"]` instead of modules params.
We cannot point to the first module parameters, as they do not have the `molecules` parameter defined in their `defaults.yaml` and therefore not valid later...
Not easy...

Reply:
In the case of `topoaa`, we have defined the parameter `molecules`, with expert level `hidden`...
An other trick